src/api/Patcher.ts

```diff
@@ -0,0 +1,18 @@
+export interface IBasePatcherApi {
+    internalId: string | undefined; /* NOT ON REPLUGGED */
+    unpatchAll(): void;
+    after<T, A = unknown[], R = unknown>(target: T, name: string, cb: (args: A, res: R, instance: T) => R): () => void;
+}
+
+class DummyPatcherApi implements IBasePatcherApi {
```

Comment:
Why is this a class and not an interface

Reply:
There already is an interface? Besides, https://github.com/LosersUnited/DiscordModCompiler/pull/7#pullrequestreview-1880151835 this

Reply:
#7 was closed because it was preatty bad, too much edits and I was new to typescript and to the project

Reply:
I'm talking about a specific comment on #7...

> The reason it was (and I think it should be) an `interface` is that so every other client mod implementation class needs to implement **each** of the methods, and now with usage of `extends` it's not needed as opposed to `implements`

Reply:
> There already is an interface?

Well..maybe in the future we should refactor that, don't see why we should need both

Reply:
> I'm talking about a specific comment on #7...
> 
> > The reason it was (and I think it should be) an `interface` is that so every other client mod implementation class needs to implement **each** of the methods, and now with usage of `extends` it's not needed as opposed to `implements`

We can do that without creating a dummy class

Reply:
> > I'm talking about a specific comment on #7...
> > > The reason it was (and I think it should be) an `interface` is that so every other client mod implementation class needs to implement **each** of the methods, and now with usage of `extends` it's not needed as opposed to `implements`
> 
> We can do that without creating a dummy class

No, TypeScript won't transpile pure interface into JavaScript which makes them get removed in dist

Reply:
And why do we need dummy classes in dist?

Reply:
how are you gonna get autocomplete if it's missing?

Reply:
Right, autoclomplete, forgot about that

Reply:
So can I mark this as resolved?